applications/RomApplication/python_scripts/hrom_training_utility.py

```diff
@@ -53,16 +57,24 @@ def AppendCurrentStepResiduals(self):
 
             if self.echo_level > 0 : KratosMultiphysics.Logger.PrintInfo("HRomTrainingUtility","RomResidualsUtility created.")
 
-        # Generate the matrix of residuals
-        if self.echo_level > 0 : KratosMultiphysics.Logger.PrintInfo("HRomTrainingUtility","Generating matrix of residuals.")
-        res_mat = self.__rom_residuals_utility.GetResiduals()
+        # Generate the matrix of projected residuals
+        if self.echo_level > 0 : KratosMultiphysics.Logger.PrintInfo("HRomTrainingUtility","Generating matrix of projected residuals.")
+        if (self.projection_strategy=="Galerkin"):
+                res_mat = self.__rom_residuals_utility.GetProjectedResidualsOntoPhi()
+        elif (self.projection_strategy=="Petrov-Galerkin"):
+                res_mat = self.__rom_residuals_utility.GetProjectedResidualsOntoPsi()
+        else: 
+            err_msg = "Projection strategy \'{}\' for hrom is not supported.".format(self.projection_strategy)
```

Comment:
```suggestion
            err_msg = f"Projection strategy \'{self.projection_strategy}\' for HROM is not supported."
```
Now we support f-strings.